source/simulator/solver.cc

```diff
@@ -855,8 +855,21 @@ namespace aspect
         solution.block(block_p) = distributed_stokes_solution.block(block_p);
 
         // print the number of iterations to screen
-        pcout << solver_control_cheap.last_step() << '+'
-              << solver_control_expensive.last_step() << " iterations.";
+        if (solver_control_cheap.last_step() == numbers::invalid_unsigned_int)
+          {
+            pcout << "0" << '+'
```

Comment:
Instead of this sequence of `if`-`else` statements, I think you can do this more concisely as
```
pcout << (solver_control_cheap.last_step() != numbers::invalid_unsigned_int ? 
                solver_control_cheap.last_step()  : 
                0)
          << '+'
          << (solver_control_expensive.last_step() != numbers::invalid_unsigned_int ? 
                solver_control_expensive.last_step()  : 
                0)
          << " iterations.";
```
This makes it easier to understand that you're not missing a case. (You only have 3 cases for the potentially four situations; the fourth can't happen because either the cheap or the expensive solver is always run, but that requires thought to determine when reading the code.)

Reply:
Definitely cleaner, thanks :+1: